### ISSUE TITLE
Can't open files in file manager using hotkeys

### ISSUE THREAD

**Issue by noah-arroyo** [Zettlr/Zettlr#4329]
### Description

**Expected behavior:** Use hotkeys to navigate the caret to the file manager, then use keystrokes to select and open a file.

**Buggy behavior:** Hotkeys _do_ navigate the caret to the file manager, but _keystrokes_ won't open a file.

### Reproducing

Press `Ctrl`+`Shift`+`t` to navigate the caret to the filter field, above the file manager. Press `↓` to highlight the first file in the list. Press `Enter`, and the file _won't_ open.

### Zettlr Version

Beta (if applicable)

### Specify version

3.0.0-beta.4

### Installation Method

From the Website or GitHub

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Windows 10

**Comment by nathanlesage** [comment]
This works for me – do you still experience the bug on the current develop branch/nightlies?

**Comment by noah-arroyo** [comment]
Still experiencing the bug.

**Comment by nathanlesage** [comment]
I've just finalized moving the main window to Vue's script setup API. This has highlighted a large bunch of potential issues. I just tried this again, feel free to have a look at Monday's nightly. (The nighly itself may still be unstable, but that specific functionality *should* work there.)

**Comment by sensologica** [comment]
I tried to reproduce the bug today but everything seems to be working correctly. I was able to successfully open files with keystrokes (macOS Sequoia 15.3.1, Zettlr 3.3.1). It works in all File Manager Display Modes (Thin, Expanded, Combined). Could you confirm if you are still experiencing this bug on newer versions of Zettlr?

I did notice one strange behavior, however.

![Image](https://github.com/user-attachments/assets/850083b1-41cf-4f68-a9b3-1f179e8fbea7)

* Suppose we start in the file `abc.md`. This means that it's pre-selected in the File Manager.
* We navigate to the filter search bar with a keystroke and type in `ab`. This filters out any irrelevant results such as `a.md` from the file list.
* The next logical step would be to navigate to `ab.md` with the `↓` arrow key. However, this action will be performed in relation to the already selected `abc.md` file and will take us to `abcd.md` instead of to `ab.md` on the top of the filtered list, which is somewhat unexpected.

**Comment by noah-arroyo** [comment]
Hi there, yes, I can confirm that on the latest Windows ver (3.4.1), the bug that I'd reported is fixed.

However, I'm also experiencing the odd behavior you've documented above. 

Personally, I'd expect that regardless of whether any file were already selected in the File Manager, hotkey-navigating the caret to the filter field and then hitting `↓` on the keyboard would highlight the File Manager's first visible file. 

Instead, hitting `↓` highlights ...

- ... the next item following the folder's name, AKA the first file visible in the list, _if no other file is already selected._ This can happen if, with Zettlr running, you close all files, then close and re-open Zettlr.
- ... the next file following an already-selected file, _if the latter is visible (not filtered out) in the File Manager._
- ... the second visible file, _if the results are filtered and no already-selected file is visible._

**Comment by nathanlesage** [comment]
So, it's been more than a year, but I finally took my time to take a look at this. The weird behavior you've seen was caused by the navigation following an entirely different tree of visible elements than what you would actually see.

Basically the following would happen:

When it comes to *displaying* files and folders, there are various settings that can show and hide files on demand. You know that already. This implies, however, that there is the *entire* directory tree somewhere in the app. Just because you can't see, say, dotfiles, they're still there (in case you decide to view them in the settings).

Now, the file list and file tree work well in this scenario because we have these humongous functions that filter out children upon rendering based on dozens of conditions.

But for navigation, the app has to treat the list of files not as a tree (as its displayed in the folder tree), but as a long, one-dimensional list of items. (This way, when you are on the last file of a folder, pressing arrow down will move to the next folder below the one you're in.)

Long story short: We never implemented any file filtering in that "virtual" list of folders. That means, if there were hidden files in between two visible ones, pressing arrow down would select that file in the virtual, unfiltered tree, and not the next actually visible one. Pressing arrow down (or up, for that matter), multiple times would eventually select the next/previous file.

Fixed in the following commit.

**Comment by noah-arroyo** [comment]
You are a gentleman and a scholar!
